tests/flow/test_foreach.py

```
@@ -0,0 +1,710 @@
+                                      [[0], [2], [4], [6], [8]])
+
+    def test06_field_access(self):
+        """validate that Foreach accesses fields correctly"""
+
+        self.graph.delete()
+
+        # create two nodes with list properties
+        self.graph.query("CREATE (:N {li: [1, 2, 3, 4]}), (:M {li: [1, 2, 3, 4]})")
+
+        # run a Foreach clause for every node, accessing its list
+        res = self.graph.query("""MATCH (n) FOREACH(i in n.li |
```

Ruff (0.12.2)

Local variable `res` is assigned to but never used

Remove assignment to unused variable `res`

(F841)